### PR TITLE
Changes trays to function more like tables

### LINKED ISSUE #10603: Food trays will not pick up any food and are currently useless.
#### Description of issue
Trays do not do anything when used on food and vice versa. They refuse to pick up any food items and are currently useless.

#### Difference between expected and actual behavior
I think that the trays should be picking up any food on a tile much like a plant bag but they do nothing at all.

#### Steps to reproduce 
Get a tray from the chef's vendor and try to pick up any food item.

#### Length of time in which bug has been known to occur

Not sure.

#### Server revision

00753e149f6c1993efc3a0d1bbb069cd7daae037 

#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [x ] Issue could be reproduced at least once
- [ ] Issue could be reproduced by different players
- [x ] Issue could be reproduced in multiple rounds
- [x ] Issue happened in a recent (less than 7 days ago) round
- [x ] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

### PR DESCRIPTION
**Pull request by JustSumBody**
closes #10603 
Click on the tray with an item and it gets placed on it like a portable table.(even works in your hand!) Putting down the tray removes all the items. I commented out the previous functionality because it was inconsistent, not sure if it's worth revisiting to have trays do both but this way is cleaner I think.

I'm not going to lie I know jack shit about what I did I just copy pasted a bunch of shit from tables and it worked out.
Tested moderately
Also this doesn't affect robotrays because honestly how are those supposed to work

### CODE REVIEW

**Comment by ComicIronic** [comment]
You should be using the proc for getting an item's weight value in the calc_carry proc.

**Comment by Shadowmech88** [comment]
> proc for getting an item's weight value

Can't you just do `item.w_class`?

**Comment by ComicIronic** [comment]
That's not what's being used in the tray code because the values aren't mapped linearly.

File: code/game/objects/items/weapons/kitchen.dm [hunk 1]
```diff
@@ -447,41 +448,76 @@
...
+
+		overlays += image
+	else
+		..()
+/obj/item/weapon/tray/proc/calc_carry() 
 	// calculate the weight of the items on the tray
 	var/val = 0 // value to return
 
 	for(var/obj/item/I in carrying)
-		if(I.w_class > W_CLASS_TINY)
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/kitchen.dm:474]: So just val += I.get_trayweight()

File: code/game/objects/items/weapons/kitchen.dm [hunk 2]
```diff
@@ -447,41 +448,72 @@
...
+		image.appearance = W.appearance
+		image.layer = W.layer + 30
+		image.plane = FLOAT_PLANE
+
+		overlays += image
+	else
+		..()
+/obj/item/weapon/tray/proc/calc_carry() 
 	// calculate the weight of the items on the tray
 	var/val = 0 // value to return
```
Comment (PJB3005) [inline code/game/objects/items/weapons/kitchen.dm]: Use `.` instead (default return value), then in the for loop:

``` DM
for(var/obj/item/I in carrying)
    . += I.get_trayweight() || INFINITY
```

**Comment by JustSumBody** [comment]
There is some funky business going on when you try to put a mob that can move on a tray(dionae nymphs), if they move off the tray their holder is still technically on it and it does a weird dupe thing. 
I don't think you were able to put a mob on a tray before, but is it worth looking into for completion's sake or just don't let it happen? 
How would you even get it to check for something like that?

**Comment by Shadowmech88** [comment]
You could put an if(istype()) check into the tray's attackby() to check for a mob holder and just disallow them from being put onto the tray.

**Comment by JustSumBody** [comment]
Right, that's what I've done for now
I'll just leave it as is then, no point in over-complicating things just to serve raw dionae nymphs

**Comment by 9600bauds** [comment]
Juan hundred percent tested?

**Comment by JustSumBody** [comment]
Money back guarantee

**Comment by 9600bauds** [comment]
Mergevalids detected

**Comment by 9600bauds** [comment]
Really damn neat now that I test it

**Comment by Kurfursten** [comment]
This is really cool stuff. I may be biased since I asked for it, but I'm glad to see it get implemented. Great work.

**Pull request merged**